samples/cooperative-cancellation/Core_9/Server/LongRunningMessageHandler.cs

```diff
@@ -1,21 +1,26 @@
 using NServiceBus;
-using NServiceBus.Logging;
+using Microsoft.Extensions.Logging;
 using System;
 using System.Threading.Tasks;
 
 public class LongRunningMessageHandler :
     IHandleMessages<LongRunningMessage>
 {
-    static ILog log = LogManager.GetLogger<LongRunningMessageHandler>();
+    private readonly ILogger<LongRunningMessageHandler> logger;
+
+    public LongRunningMessageHandler(ILogger<LongRunningMessageHandler> logger)
+    {
+        this.logger = logger;
+    }
 
     #region LongRunningMessageHandler
     public async Task Handle(LongRunningMessage message, IMessageHandlerContext context)
     {
-        log.Info($"Received message {message.DataId}. Entering loop.");
+        logger.LogInformation($"Received message {message.DataId}. Entering loop.");
```

Comment:
use structured logging

Isn't the compiler providing you a hint that string interpolation should not be used for logging?